### PR TITLE
Revert "[analyzer] Use absolute path to logger.so in LD_PRELOAD"

### LINKED ISSUE #4312: CodeChecker log does not work for 32bits binaries on 64bits platform
Logging is successful on a 64 bits binary:
```
CodeChecker` log -b "./builder64" -o ./compile_command64.json
[INFO 2024-08-14 14:09] - Starting build...
[INFO 2024-08-14 14:09] - Using CodeChecker ld-logger.
[INFO 2024-08-14 14:09] - Build finished successfully.
(CodeChecker venv-dev) ednikru@seliiuvd02700[14:09][workspace/test_files/codechecker_loggin_test]$ cat compile_command64.json 
[
        {
                "directory": "/local/workspace/test_files/codechecker_loggin_test",
                "command": "/usr/bin/gcc ./test.c",
                "file": "./test.c"
        }
]
```

And it fails for 32 bit binary (on a 64 bits machine):
```
(CodeChecker venv-dev) ednikru@seliiuvd02700[14:10][workspace/test_files/codechecker_loggin_test]$ CodeChecker log -b "./builder32" -o ./compile_command32.json
[INFO 2024-08-14 14:10] - Starting build...
[INFO 2024-08-14 14:10] - Using CodeChecker ld-logger.
ERROR: ld.so: object '/local/workspace/codechecker/build/CodeChecker/ld_logger/lib/x86_64/ldlogger.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
[INFO 2024-08-14 14:10] - Build finished successfully.
`
builder.c
`
#include<stdlib.h>
int main(void){
  const char* command="gcc ./test.c";
  system(command);
  return 0;
}
```

`gcc ./builder.c -m32 -o builder32`

### PR DESCRIPTION
**Pull request by dkrupp**
This reverts commit db924b02424c4ed5fc5c0b21af19c848ad45e6f2.

ld_logger.so binary version (32 or 64 bits) must depend on the excuted builder binary version, not on the host machine platform version.

Fixes #4312

### CODE REVIEW

File: analyzer/tools/build-logger/README.md [hunk 1]
```diff
@@ -14,8 +14,9 @@ make all test
 
 Set the following environment variables:
 ~~~~~~~
-export LD_PRELOAD=`pwd`/build/lib/`uname -m`/ldlogger.so
-export CC_LOGGER_GCC_LIKE="gcc:g++:clang:clang++:/cc:c++"
+export LD_PRELOAD=ldlogger.so
+export LD_LIBRARY_PATH=`pwd`/build/lib:$LD_LIBRARY_PATH
+export CC_LOGGER_GCC_LIKE="gcc:g++:clang:clang++:cc:c++"
```
Comment (bruntib) [inline analyzer/tools/build-logger/README.md:19]:
```suggestion
export CC_LOGGER_GCC_LIKE="gcc:g++:clang:clang++:/cc:c++"
```

File: analyzer/tools/build-logger/tests/unit/__init__.py [hunk 2]
```diff
@@ -106,14 +106,11 @@ def read_actual_json(self) -> str:
...
         return {
             "PATH": os.getenv("PATH"),
-            "LD_PRELOAD": os.path.join(LOGGER_DIR,
-                                       "lib",
-                                       machine,
-                                       "ldlogger.so"),
-            "CC_LOGGER_GCC_LIKE": "gcc:g++:clang:clang++:/cc:c++",
+            "LD_PRELOAD": "ldlogger.so",
+            "LD_LIBRARY_PATH": os.path.join(LOGGER_DIR, "lib"),
+            "CC_LOGGER_GCC_LIKE": "gcc:g++:clang:clang++:cc:c++",
```
Comment (bruntib) [inline analyzer/tools/build-logger/tests/unit/__init__.py:113]:
```suggestion
            "CC_LOGGER_GCC_LIKE": "gcc:g++:clang:clang++:/cc:c++",
```

**Review by bruntib — CHANGES_REQUESTED** [review]
There are some Pylint warnings too, otherwise it looks good.

**Comment by dkrupp** [comment]
thanks for the review. pylint problems also fixed.

**Review by bruntib — APPROVED** [review]

**Comment by odin-** [comment]
Great :-). Thanks for the fix.

**Pull request merged**